### DISCUSSION TITLE
0.14?

### DISCUSSION THREAD

**Discussion by ElZorroofOz** [bimmerconnected/bimmer_connected#672] [Q&A] [answered]
Just installed on a Pi4 and a Pi3 and the version downloaded is 0.14?
It works despite Python being 3.7, but I'm not sure if it's not the reason behind the other error (HTTP401)

**Comment by rikroe** [comment, marked as answer]
You need to update your python version or install an additional python version like 3.10+.
There's tutorials for that.

Make sure not to replace, but install additionally.

**Reply by ElZorroofOz** [reply]
Thanks. I do shiver at the thought of doing that on a pi3 that is used for real time decision making for the health of our energy system (we're in remote australia, without solar... cold showers).

**Reply by ElZorroofOz** [reply]
... but will try when I can muster the time. Thanks for the answer, though.

**Reply by rikroe** [reply]
I can understand that. Definitely check python virtual environments and/or python altinstall.
This should keep your original python intact and you can split your requirements better.

**Reply by ElZorroofOz** [reply]
@rikroe Is there anything specific in the install of Python 3.12 that I need to do to ensure that pip3 will download 0.16 from bimmerconnected? (I'd like to keep 3.7 as the standard version since I have old scripts reliant on it that I wouldnt even know where to find...). Thanks !

**Reply by rikroe** [reply]
You can follow this guide here: https://aruljohn.com/blog/python-raspberrypi/
Just don't do the step that configures Python 3.12 as default version. 

Other than having Python 3.12 and using pip inside the 3.12 environment, either using a venv (recommended) or `python3.12 -m pip install bimmer_connected`, you should be good.
